### PR TITLE
78 artists page server

### PR DESCRIPTION
**Pull request by Eyal-donut**
-  created an endpoint for loading artists data
 - create a function that takes artist name and returns artist data. connected it to the endpoint
 - query mongodb artist collection based on names (3 languages!)
 - load ALL songs of this artist (they are populated in the artist document in the response)
- wrote unit tests for the function and the endpoint

### CODE REVIEW

File: server/scrapping/scrapingArabicLyrics.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
+import puppeteer from "puppeteer"
```
Comment (yishain11) [inline server/scrapping/scrapingArabicLyrics.js:1]: why is this a part of the PR?

File: server/scrapping/reformatScrapingInput.js [hunk 2]
```diff
@@ -0,0 +1,6 @@
+const reformatScrapingInput = (firstInput, secondInput) => {
```
Comment (yishain11) [inline server/scrapping/reformatScrapingInput.js:1]: why is this a part of the PR?

File: server/test/getArtists.tests.js [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+const expect = chai.expect;
+
+
+
+describe('getArtists', function () {
+  it('Should check getArtists endpoint and fetched data', async function () {
+
+    this.timeout(10000);
+
+    const requestBodies = [{"name": {"hebrew" : "אמן"}}, {"name": {"english" : "Eyal"}}, {"name": {"english" : "moshE"}}, {"name": {"arabic" : "فنان"}}]
```
Comment (yishain11) [inline server/test/getArtists.tests.js:14]: this tests assumes that there is such an object. you need to create this object as a part of "before" and delete it in "after"

File: server/utils/createObjectFromQuery.js [hunk 4]
```diff
@@ -7,7 +7,7 @@ const createObjectFromQuery = (data) => {
         filteredValues['name.hebrew'] = name.hebrew
       }
       if (name.english) {
-        filteredValues['name.english'] = name.english
+        filteredValues['name.english'] = new RegExp(`^${name.english}`, 'i');
```
Comment (yishain11) [inline server/utils/createObjectFromQuery.js:10]: why not use includes?

**Review by yishain11 — CHANGES_REQUESTED** [review]

File: server/scrapping/scrapingArabicLyrics.js (hunk 1 above)
Comment (Eyal-donut) [reply server/scrapping/scrapingArabicLyrics.js:1]: This is a separate PR - see PR 69 scraping arabic lyrics

File: server/test/getArtists.tests.js (hunk 3 above)
Comment (Eyal-donut) [reply server/test/getArtists.tests.js]: I updated the test accordingly, but I don't understand why to do that.

File: server/utils/createObjectFromQuery.js (hunk 4 above)
Comment (Eyal-donut) [reply server/utils/createObjectFromQuery.js:10]: Here I am creating on object, that I then can pass as a filter in any find() or findOne() or findOneAndUpdate() of Mongoose, to query by this object.
I define here the fields of the object 'filteredValues' and not making any comparison.

**Review by yishain11 — CHANGES_REQUESTED** [review]

**Comment by Eyal-donut** [comment]
All merge conflicts resolved

**Pull request closed without merging**